connection.js

```diff
@@ -614,4 +623,24 @@ function sendLazyErrorFrame(reqFrame, codeString, message) {
         codeString, message);
 };
 
+TChannelConnection.prototype._drain =
+function _drain(reason, exempt) {
+    var self = this;
+
+    TChannelConnectionBase.prototype._drain.call(self, reason, exempt);
+
+    if (self.remoteName) {
+        sendDrainingFrame();
+    } else {
+        self.identifiedEvent.on(sendDrainingFrame);
+    }
+
+    function sendDrainingFrame() {
+        self.handler.sendErrorFrame(
+            v2.Frame.NullId, null,
+            'Declined',
```

Comment:
Declined, this doesn't indicate any particular error or problem.  Think of it as a "promise to decline and eventually close", really we want a new error code for this, but are past the point of adding a new one to v2 as far as I see it.